src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -759,5 +763,153 @@ public Task CommitFilesAsync(List<GitFile> filesToCommit, string repoUri, string
         {
             return this.CommitFilesAsync(filesToCommit, repoUri, branch, commitMessage, _logger, _personalAccessToken);
         }
+
+        /// <summary>
+        ///     Add the informed build as an specific build artifact source to the release definition informed.
+        /// </summary>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="releaseDefinition">Release definition to be updated</param>
+        /// <param name="build">Build which should be added as source of the release definition.</param>
+        /// <returns>AzureDevOpsReleaseDefinition</returns>
+        public async Task<AzureDevOpsReleaseDefinition> AddArtifactSourceAsync(string accountName, string projectName, AzureDevOpsReleaseDefinition releaseDefinition, AzureDevOpsBuild build)
```

Comment:
I think I'm safe to assume that this doesn't work across azure devops instances? E.g. build in DevDiv being used as a source for a release pipeline in dnceng.

Not a big deal if it does, just might mean that DotNet-Trusted and a couple other repos may not have the new flow enabled until they move to dnceng.  Alternately we could create a parallel release pipeline over there.

Reply:
Yeah, I think AzDO doesn't support that.